modules/utils.py

```diff
@@ -0,0 +1,143 @@
+import hashlib
+import os
+from datetime import datetime
+
+import dateutil.parser
+
+from modules import constants
+
+
+def remove_empty_keys(d):
+    return {k: v for k, v in d if v is not None}
+
+
+def get_url_hash(url):
+    return int(hashlib.md5(url.encode()).hexdigest()[0:4], 16)
+
+
+def get_ext(ext):
+    ret_ext = ext
+    i_q = ret_ext.find('?')
+    if i_q > -1:
+        ret_ext = ret_ext[:i_q]
```

Comment:
I wonder should we filter out hash urls as well?

e.g. https://www.example.com/index.html#1234.1234

Reply:
Resolving per other comment